packages/clerk-js/src/core/clerk.ts

```diff
@@ -1160,48 +1153,78 @@ export default class Clerk implements ClerkInterface {
     }
 
     const satelliteUrl = getClerkQueryParam(CLERK_SATELLITE_URL);
-    if (!satelliteUrl) {
-      return false;
-    }
-
-    if (!isHttpOrHttps(satelliteUrl)) {
-      clerkRedirectUrlIsMissingScheme();
-    }
-
-    return true;
+    return !!satelliteUrl;
   };
 
   #syncWithPrimary = async () => {
-    if (this.#shouldSyncWithPrimaryInDevelopment()) {
+    if (this.instanceType === 'development') {
       await this.navigate(this.#buildSyncUrlForDevelopmentInstances());
-    } else if (this.#shouldSyncWithPrimaryInProduction()) {
+    } else if (this.instanceType === 'production') {
       await this.navigate(this.#buildSyncUrlForProductionInstances());
     }
   };
 
+  #validateMultiDomainOptions = () => {
+    if (!this.isSatellite) {
+      return;
+    }
+    if (this.#instanceType === 'development' && !this.#options.signInUrl) {
+      clerkMissingSignInUrlAsSatellite();
+    }
+
+    if (!this.proxyUrl && !this.domain) {
+      clerkMissingProxyUrlAndDomain();
+    }
+  };
+
   #loadInStandardBrowser = async (): Promise<boolean> => {
-    // Dev Browser handling
+    /**
+     * 1. Create the devBrowser.
+     * At this point the devBrowser is not yet setup, but its API is ready for use
```

Comment:
@panteliselef let's explain why this step is required in the comment. If not mistake the devBrowserHandler is used by some multi-domain helpers. This should be stated explicitly.